Blocks/URL/url.h

```diff
@@ -254,6 +257,55 @@ struct URLParametersExtractor {
     }
     const std::map<std::string, std::string>& AsImmutableMap() const { return parameters_; }
     std::map<std::string, std::string> parameters_;
+
+    // NOTE: `FillObject` only populates the fields present in the URL; it doesn't erase what's not in the querystring.
+    template <typename T>
+    const T& FillObject(T& object) const {
+      QueryParametersObjectFiller parser{parameters_};
+      current::reflection::VisitAllFields<T, current::reflection::FieldNameAndMutableValue>::WithObject(object, parser);
+      return object;
+    }
+
+    template <typename T>
+    T FillObject() const {
+      T object;
+      FillObject(object);
+      return object;
+    }
+
+   private:
+    struct QueryParametersObjectFiller {
+      const std::map<std::string, std::string>& q;
+
+      // The `std::string` is a special case, as no quotes are expected.
+      void operator()(const std::string& key, std::string& value) const {
+        const auto cit = q.find(key);
+        if (cit != q.end()) {
+          value = cit->second;
+        }
+      }
+
+      // The `Optional<std::string>` is also a special case, as no quotes are expected.
+      void operator()(const std::string& key, Optional<std::string>& value) const {
+        const auto cit = q.find(key);
+        if (cit != q.end()) {
+          value = cit->second;
+        }
+      }
+
+      // For the remaining field types, use `ParseJSON`. Overkill, but ensures any `CURRENT_STRUCT` can be URL-encoded.
+      template <typename T>
+      void operator()(const std::string& key, T& value) const {
+        const auto cit = q.find(key);
+        if (cit != q.end()) {
+          try {
+            ParseJSON(cit->second, value);
+          } catch (const current::TypeSystemParseJSONException&) {
+            // Silently discard the exception.
```

Comment:
I'm afraid it would make the logic trickier -- there's no simple way for URL parsing to trigger an error.

In my head, passing complex data in URL parameters is a bad style that's discouraged, and, under this assumption, the user who's doing that knows what are they doing. Keep in mind, this method extends the object, doesn't fully parse it.